src/widget/wsearchlineedit.cpp

```diff
@@ -223,22 +225,39 @@ QString WSearchLineEdit::getSearchText() const {
 }
 
 void WSearchLineEdit::focusInEvent(QFocusEvent* event) {
+#if ENABLE_TRACE_LOG
+    kLogger.trace()
+            << "focusInEvent";
+#endif // ENABLE_TRACE_LOG
     QLineEdit::focusInEvent(event);
     showSearchText(getSearchText());
 }
 
 void WSearchLineEdit::focusOutEvent(QFocusEvent* event) {
+#if ENABLE_TRACE_LOG
+    kLogger.trace()
+            << "focusOutEvent";
```

Comment:
What is the value of adding this? Why not just set a breakpoint in a debugger if you're interested in this information?

Reply:
I might need to see the sequence of events without losing focus.

Reply:
Switching to a different windows is not what you want then.

Reply:
I'll keep the trace log code just in case we need this again.